internal/service/directconnect/connection.go

```diff
@@ -40,6 +41,13 @@ func ResourceConnection() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validConnectionBandWidth(),
 			},
+			// The MAC Security (MACsec) connection encryption mode.
+			"encryption_mode": {
+				Type:         schema.TypeString,
+				Computed:     true,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"no_encrypt", "should_encrypt", "must_encrypt"}, false),
```

Comment:
Please consider removing validation, as this is a computed field. If the API is updated with more statuses in the future, validating against these 3 values could cause problems.
```suggestion

```

Reply:
Encryption mode is both returned by the API and an optional parameter that can be provided by the user (e.g. to change mode). The input validation is required for when the user applies this parameter. Might make more sense to remove the `Computed` attribute in this case. Ref: https://docs.aws.amazon.com/sdk-for-go/api/service/directconnect/#UpdateConnectionInput